src/tss2-policy/tss2_policy.c

```diff
@@ -0,0 +1,124 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*******************************************************************************
+ * Copyright 2018-2019, Fraunhofer SIT sponsored by Infineon Technologies AG
+ * All rights reserved.
+ *******************************************************************************/
+
+#ifdef HAVE_CONFIG_H
+#include <config.h>
+#endif
+
+#include <string.h>
+#include <stdlib.h>
+
+#include "ifapi_policy_calculate.h"
+#include "ifapi_policy_execute.h"
+#include "ifapi_policy_instantiate.h"
+#include "ifapi_policy_json_deserialize.h"
+#include "tpm_json_deserialize.h"
+
+#include "util/aux_util.h"
+
+TSS2_RC
+Tss2_PolicyExecute(
+    TPMI_ALG_HASH hash_alg,
+    TPMS_POLICY *policy,
+    ESYS_CONTEXT *esys_ctx,
+    ESYS_TR session)
+{
+    IFAPI_POLICY_EXEC_CTX context = { 0 };
+
+    TSS2_RC r = ifapi_policyeval_execute_prepare(&context, hash_alg, policy);
+    if (r) {
+        return 1;
+    }
+
+    context.session = session;
+
+    return ifapi_policyeval_execute(esys_ctx, &context);
+}
+
+TSS2_RC
+Tss2_PolicyCalculate(
+        TPML_POLICYELEMENTS *policy,
+        TPML_DIGEST_VALUES *policyDigests,
+        TPMI_ALG_HASH hash_alg,
+        size_t hash_size,
+        size_t digest_idx)
```

Comment:
@AndreasFuchsSIT can you enlighten me as to what hash_size and digest_idx do, this is calling ifapi_calculate_policy, and I can't figure out what these inputs do. Also, why is hash_size needed if hash_alg is passed?

Reply:
Oh I get digest_idx, I see that it's a list of policies that get passed.

Reply:
So can a json file have N>1 policies contained in it?

Reply:
Kinda works, the FAPI async model is messing with me on execute. I guess I would have to do a little more work in the internals so make a blocking version of FAPI policy execution.

Reply:
If I remember correctly, it's as follows:
A policy can have multiple digests, one for each hash-alg.
Thus the list of digests and the array-index we give here.

Reply:
Yes e.g. Fapi_ExportPolicy computes the policy digest for every name alg used in the current fapi profiles if possible.

Reply:
Ok cool, so I think in this regard we just use index 0, is that always the hashalg specified by the caller of ifapi_calculate_policy?

Reply:
yes, the caller decides which hashAlg goes into which index.